### PR TITLE
add single keep for extension (given libraryId)

### PR DESCRIPTION
**Pull request by ahsu1230**
@2is10 @andrewconner 

also checks if valid membership to libraryId

### CODE REVIEW

File: kifi-backend/modules/shoebox/app/com/keepit/controllers/ext/ExtLibraryController.scala [hunk 1]
```diff
@@ -32,4 +38,29 @@ class ExtLibraryController @Inject() (
     Ok(Json.obj("libraries" -> Json.toJson(jsons)))
   }
 
+  def addKeep(pubId: PublicId[Library]) = JsonAction.authenticatedParseJson { request =>
+    val idTry = Library.decodePublicId(pubId)
+    idTry match {
+      case Failure(ex) =>
+        BadRequest(Json.obj("error" -> "invalid id"))
```
Comment (andrewconner) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/ext/ExtLibraryController.scala]: Mind switching to something like "invalid_library_id"? This is never displayed, and often strings with spaces seem like error messages. Most of our other endpoints use snake case tokens, @2is10 feel free to correct me if you disagree

File: kifi-backend/modules/shoebox/test/com/keepit/controllers/ext/ExtLibraryControllerTest.scala [hunk 2]
```diff
@@ -78,7 +78,85 @@ class ExtLibraryControllerTest extends Specification with ShoeboxTestInjector wi
...
+              |"id":"${keep1.externalId}",
+              |"title":"kayne-fidence",
+              |"url":"http://www.imagenius.com",
+              |"isPrivate":false,
+              |"libraryId":"${pubId1.id}"}""".stripMargin
+        ))
+
+        val request2 = FakeRequest("POST", path2).withBody(
+          Json.obj(
+            "title" -> "IMMA LET YOU FINISH",
```
Comment (ahsu1230) [inline kifi-backend/modules/shoebox/test/com/keepit/controllers/ext/ExtLibraryControllerTest.scala:136]: shots fired!

File: kifi-backend/modules/shoebox/test/com/keepit/controllers/ext/ExtLibraryControllerTest.scala [hunk 3]
```diff
@@ -78,7 +78,85 @@ class ExtLibraryControllerTest extends Specification with ShoeboxTestInjector wi
...
+        status(result2) must equalTo(OK)
+        contentType(result2) must beSome("application/json")
+        val keep2 = db.readOnlyMaster { implicit s => keepRepo.getByLibrary(lib2.id.get, 10, 0).head }
+        Json.parse(contentAsString(result2)) must equalTo(Json.parse(
+          s"""
+            |{
+              |"id":"${keep2.externalId}",
+              |"title":"IMMA LET YOU FINISH",
+              |"url":"http://www.beyonceisbetter.com",
+              |"isPrivate":true,
```
Comment (andrewconner) [inline kifi-backend/modules/shoebox/test/com/keepit/controllers/ext/ExtLibraryControllerTest.scala:149]: You can remove this. New endpoints should ignore this completely.

Comment (andrewconner) [inline kifi-backend/modules/shoebox/test/com/keepit/controllers/ext/ExtLibraryControllerTest.scala:149]: Ignore, you're right. This is output, my mistake.

**Comment by 2is10** [comment]
:+1:

**Pull request merged**